lambda_function/src/file_processor/file_processor.py

```diff
@@ -169,6 +174,35 @@ def __init__(
                 }
             )
 
+        secret_arn = os.getenv("RDS_SECRET_ARN", None)
+        if secret_arn:
+            try:
+                # Get Database Credentials
+                session = boto3.session.Session()
+                client = session.client(service_name="secretsmanager")
+                response = client.get_secret_value(SecretId=secret_arn)
+                secret = json.loads(response["SecretString"])
+                connection_string = (
+                    f"postgresql://{secret['username']}:{secret['password']}@"
+                    f"{secret['host']}:{secret['port']}/{secret['dbname']}"
+                )
+                # Initialize the database engine
+                self.database_engine = create_engine(connection_string)
+                # Setup the database tables if they do not exist
+                create_tables(self.database_engine)
+                # Set tracker to CDFTracker
+                self.tracker = tracker.CDFTracker(
+                    self.database_engine, science_filename_parser
```

Comment:
Passing the `parser` as an arg so it can be used outside HERMES?

Reply:
Yeah, exactly, so someone can build their own parsing function as well even if they don't use HERMES or our template.